lib/livebook/runtime/node_pool.ex

```diff
@@ -0,0 +1,84 @@
+defmodule Livebook.Runtime.NodePool do
+  use GenServer, async: false
+
+  # A pool for reusing child node names.
+  #
+  # `pool` refers to the list of unused names
+  #
+  # `buffer_time` refers to time the pool waits before
+  # adding a name to `pool`,  which by default is 1 minute.
+
+  @default_time 6_000
+
+  # Client interface
+
+  @doc """
+  Starts the node pool with the buffer time as 1 minute.
+  """
+  def start do
+    GenServer.start(__MODULE__, @default_time, name: __MODULE__)
+  end
+
+  @doc """
+  Starts the node pool with the buffer time as input.
+  """
+  def start(buffer_time) do
+    GenServer.start(__MODULE__, buffer_time, name: __MODULE__)
+  end
+
+  @doc """
+  Retuns a node name.
+
+  Generates a new one if node is empty, or takes on from pool.
+  """
+  def get_name(parent) do
```

Comment:
Nit: since this is pretty independent now, something like `basename` would be probably more clear than `parent`.

Reply:
Sure.